app/models/description.rb

```diff
@@ -400,25 +400,46 @@ def group_users(table)
   # Array of ids.  Caches result.
   def group_user_ids(table)
     @group_user_ids ||= {}
-    @group_user_ids[table] ||= self.class.connection.select_values(%(
-      SELECT DISTINCT u.user_id FROM #{table} t, user_groups_users u
-      WHERE t.#{type_tag}_id = #{id}
-        AND t.user_group_id = u.user_group_id
-      ORDER BY u.user_id ASC
-    )).map(&:to_i)
+    select_manager = arel_select_group_user_ids(table)
+    @group_user_ids[table] ||= self.class.connection.
+                               select_values(select_manager.to_sql).map(&:to_i)
   end
 
   # Do minimal query to enumerate a list of groups.  Return as an Array of ids.
   # Caches result.  (Equivalent to using <tt>association.ids</tt>, I think.)
   def group_ids(table)
     @group_ids ||= {}
-    @group_ids[table] ||= self.class.connection.select_values(%(
-      SELECT DISTINCT user_group_id FROM #{table}
-      WHERE #{type_tag}_id = #{id}
-      ORDER BY user_group_id ASC
-    )).map(&:to_i)
+    select_manager = arel_select_group_ids(table)
+    # puts(select_manager.to_sql)
+    @group_ids[table] ||= self.class.connection.
+                          select_values(select_manager.to_sql).map(&:to_i)
```

Comment:
Do we need `.map(&:to_i)`? 
(I'm having trouble concentrating on this, but it looks to me like `arel_select_group_ids` will always return integers.)